### DISCUSSION TITLE
white sheet

### DISCUSSION THREAD

**Discussion by Red-DawnX** [jeremyckahn/chitchatter#30] [Q&A]
I can log into the room from localhost without any problems. but when I connect with the ip address, the home page comes up, when I want to enter the room, the white page appears.

**Comment by jeremyckahn** [comment]
Oh no! Thanks for letting me know about this @Red-DawnX. The white screen means that there was an error. Chitchatter doesn't have very good error handling right now, and that's something I'll need to address soon.

> when I connect with the ip address

I'm not sure that I'm following what you mean here. Are you referring to connecting via https://chitchatter.im/?

It would be helpful for me to know if there's anything in your browser's developer console. Do you know how to access that and share any error messages that might be there when the error occurs?

**Comment by Red-DawnX** [comment]
![Ekran Alıntısı](https://user-images.githubusercontent.com/114663101/193088758-42c1fe63-fb83-4366-8197-7d2c40be9ddd.PNG)

It happens when I try to login with the ip address here. I'm just wondering why this error occurs. I've been trying for a long time

**Reply by jeremyckahn** [reply]
It looks like you're trying to access a locally-hosted Chitchatter instance via your LAN IP. I see this error in my browser console when I do the same:

```
Uncaught TypeError: crypto.subtle is undefined
    joinRoom torrent.js:49
    initGuard utils.js:44
    PeerRoom PeerRoom.ts:10
    <anonymous> usePeerRoom.ts:7
    React 3
    usePeerRoom usePeerRoom.ts:7
    Room Room.tsx:35
    React 16
    call bundle.js:41741
    call bundle.js:41740
    applyTx bundle.js:41195
    push bundle.js:41224
    navigate hooks.tsx:185
    handleFormSubmit Home.tsx:39
    React 15
```

This seems to be due to an implementation detail of [Trystero](https://github.com/dmotz/trystero/blob/af43c20eec2f5178202a0bb313a75072352d68fa/src/crypto.js#L14-L24), the library that Chitchatter uses for peer communication. Trystero is using [`Crypto.subtle`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle), but that API is only available in [secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts). In other words, we're seeing a failure because the LAN IP is neither `localhost` nor served via `https`.

**Reply by Red-DawnX** [reply]
Thank you very much, I was able to upload it.

**Reply by jeremyckahn** [reply]
Awesome, I'm glad to hear that! Out of curiosity, does this mean that you are hosting your own Chitchatter instance?

**Reply by Red-DawnX** [reply]
yes I want to try a few little things to improve myself.

**Reply by jeremyckahn** [reply]
Very cool. If you wind up with any changes you think others might benefit from, please open a PR to get it merged upstream! 🙂
